diff --git a/node_modules/riffy/build/structures/Node.js b/node_modules/riffy/build/structures/Node.js
index bac9f07..c9be0c0 100644
--- a/node_modules/riffy/build/structures/Node.js
+++ b/node_modules/riffy/build/structures/Node.js
@@ -18,7 +18,6 @@ class Node {
     this.sessionId = node.sessionId || null;
     this.rest = new Rest(riffy, this);
     Object.defineProperty(this, "options", {
-      writable: false,
       get() {
         return options
       }